src/vs/workbench/contrib/chat/browser/chatMessageUrlHandler.ts

```diff
@@ -0,0 +1,209 @@
+			}
+
+			await chatWidget.waitForReady();
+
+			await this.commandService.executeCommand(ACTION_ID_NEW_CHAT);
+			await this.commandService.executeCommand(CHAT_OPEN_ACTION_ID, opts);
+
+			return true;
+		} catch (error) {
+			this.logService.error(`[ChatMessageUrlHandler] Error handling chat message URL ${uri.toString()}`, error);
+			return true;
+		}
+	}
+
+	private async shouldBlockChatMessage(messageText: string): Promise<boolean> {
+		const previewText = messageText.length > PREVIEW_MESSAGE_CHAR_LENGTH
+			? messageText.substring(0, PREVIEW_MESSAGE_CHAR_LENGTH) + '...'
+			: messageText;
+
+		const markdown = new MarkdownString([
+			localize('confirmChatMessageDetail', "This will insert the following text into chat:\n\n"),
+			`\n${previewText}\n`,
+			localize('confirmChatMessageSecurity',
+				"\n\nIf you did not initiate this request, it may represent an attempted attack on your system. Unless you took an explicit action to initiate this request, you should not continue.")
+		].join(''), { supportHtml: true });
+
+		const { confirmed } = await this.dialogService.confirm({
+			type: 'info',
+			primaryButton: localize({ key: 'yesButton', comment: ['&& denotes a mnemonic'] }, "&&Yes"),
+			cancelButton: localize('noButton', "No"),
+			message: localize('confirmChatMessage', "An external application wants to insert text into chat. Do you want to continue?"),
+			custom: {
+				markdownDetails: [{ markdown }]
+			}
+		});
+
+		return !confirmed;
+	}
+
+	/**
+	 * Cleans prompt text by removing invisible characters, control sequences, and normalizing whitespace.
+	 * Uses VS Code's existing utilities for comprehensive and consistent text sanitization.
+	 * Also removes potentially deceptive URLs (homograph attacks) for security.
+	 */
+	private sanitizePromptText(text: string): string {
+		// First remove ANSI escape sequences (terminal control codes)
+		text = removeAnsiEscapeCodes(text);
+
+		// Remove invisible characters using the comprehensive VS Code utility
+		let cleanText = '';
+		for (let i = 0; i < text.length; i++) {
+			const codePoint = text.codePointAt(i);
+			if (codePoint !== undefined) {
+				// Keep the character if it's not invisible (except for regular spaces)
+				if (!InvisibleCharacters.isInvisibleCharacter(codePoint) || codePoint === 32 /* space */) {
+					cleanText += String.fromCodePoint(codePoint);
+					// Skip the next character if this was a surrogate pair
+					if (codePoint > 0xFFFF) {
+						i++;
+					}
+				}
+			}
+		}
+
+		// Check for potentially deceptive URLs (homograph attacks)
+		cleanText = this.detectAndWarnAboutSuspiciousUrls(cleanText);
+
+		// Normalize multiple whitespace to single spaces and trim
+		return cleanText.replace(/\s+/g, ' ').trim();
+	}
+
+	/**
+	 * Detects URLs with potentially deceptive characters (homograph attacks) and removes them.
+	 * Replaces suspicious URLs with a warning message for security.
+	 */
+	private detectAndWarnAboutSuspiciousUrls(text: string): string {
```

Comment:
Should this connect to the allow list of URLs we have when opening links to only trust those?